### PR TITLE
feat(Attendance): Add In and Out time to attendance

### LINKED ISSUE #20881: Feature : Checkin Register
Dear Rock Developer.  @karthikeyan5  @Vigneshsekar 

We have a request for you, kindly add a report for the Employee Check-in Register As per below format.

image

### PR DESCRIPTION
**Pull request by Vigneshsekar**
This PR adds In time and Out time when the attendance is marked using auto-attendance.

screenshot

Once this is merged, this report (https://github.com/frappe/erpnext/issues/20881) can be built using the report builder.

Closes #20881

### CODE REVIEW

File: erpnext/hr/doctype/attendance/attendance.json [hunk 1]
```diff
@@ -172,13 +176,34 @@
...
+  },
+  {
+   "fieldname": "details_section",
+   "fieldtype": "Section Break",
+   "label": "Details"
+  },
+  {
+   "fieldname": "in_time",
+   "fieldtype": "Datetime",
+   "label": "In Time",
```
Comment (Anurag810) [inline erpnext/hr/doctype/attendance/attendance.json:189]: The display should depend on shift if  ```shift != None```

File: erpnext/hr/doctype/attendance/attendance.json [hunk 2]
```diff
@@ -172,13 +176,34 @@
...
+   "read_only": 1
+  },
+  {
+   "fieldname": "out_time",
+   "fieldtype": "Datetime",
```
Comment (Anurag810) [inline erpnext/hr/doctype/attendance/attendance.json:195]: The display should depend on shift if ```shift != None```

File: erpnext/hr/doctype/attendance/attendance.json [hunk 3]
```diff
@@ -172,13 +176,34 @@
    "fieldname": "early_exit",
    "fieldtype": "Check",
```
Comment (Anurag810) [inline erpnext/hr/doctype/attendance/attendance.json:177]: Early exit and late entry should be calculated using In and Out time. Using field `start_time` and `end_time` in Shift type document.

**Review by Anurag810 — CHANGES_REQUESTED** [review]

File: erpnext/hr/doctype/attendance/attendance.json (hunk 3 above)
Comment (karthikeyan5) [reply erpnext/hr/doctype/attendance/attendance.json:177]: Yes, this already happens here: https://github.com/frappe/erpnext/blob/develop/erpnext/hr/doctype/shift_type/shift_type.py#L43

File: erpnext/hr/doctype/attendance/attendance.json (hunk 2 above)
Comment (karthikeyan5) [reply erpnext/hr/doctype/attendance/attendance.json:195]: the display depends on has been added.

File: erpnext/hr/doctype/attendance/attendance.json (hunk 1 above)
Comment (karthikeyan5) [reply erpnext/hr/doctype/attendance/attendance.json:189]: the display depends on has been added.

**Review by Anurag810 — APPROVED** [review]

**Comment by karthikeyan5** [comment]
@Mergifyio backport version-13-beta-pre-release

**Pull request merged**